### PR TITLE
stop using buildkit hack to pull images

### PR DESCRIPTION
**Pull request by ndeloof**
**What I did**
Stop using Buildkit hack to get image pulled, first pull then build
TODO investigate how we could get those in parallel within a shared progress.writer

**Related issue**
close https://github.com/docker/compose-cli/issues/1732

**(not mandatory) A picture of a cute animal, if possible in relation with what you did**

### CODE REVIEW

**Comment by gtardif** [comment]
This should also allow better error metrics, separating pull issues from build issues

File: local/compose/build.go [hunk 1]
```diff
@@ -87,7 +86,18 @@ func (s *composeService) Build(ctx context.Context, project *types.Project, opti
 }
 
 func (s *composeService) ensureImagesExists(ctx context.Context, project *types.Project, observedState Containers, quietPull bool) error {
-	images, err := s.getImageDigests(ctx, project)
+	for _, service := range project.Services {
+		if service.Image == "" && service.Build == nil {
+			return fmt.Errorf("invalid service %q. Must specify either image or build", service.Name)
```
Comment (mat007) [inline local/compose/build.go:91]: nit: I think go errors are not supposed to contain caps, what about
```suggestion
			return fmt.Errorf("invalid service %q: must specify either image or build", service.Name)
```
?

**Review by mat007 — APPROVED** [review]
Makes sense!

File: local/compose/build.go (hunk 1 above)
Comment (ndeloof) [reply local/compose/build.go:91]: AFAIk there's a few other places we didn't strictly applied this gopherextremism.
Considering this error was just moved by this PR, to be debated in a separate PR if we want to also apply this "rule"

**Pull request merged**